### PR TITLE
Refactor/list optimization

### PR DESCRIPTION
**Pull request by hujiahao-hjh**
## What this PR does / why we need it:
List optimization and improved cluster list display.

## Does this PR introduce a user interface change?

✅ Yes(screenshot is required)
image

## ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | List optimization and improved cluster list display. |
| 🇨🇳 中文    | 优化列表以及改善集群列表的展现形式。 |

## Does this PR need be patched to older version?
❎ No

## Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: shell/app/common/components/list/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import ListItem from 'common/components/list/list-item';
+import './index.scss';
+
+const List = (props: ERDA_LIST.IProps) => {
+  const { dataSource, isLoadMore, onLoadMore, pagination, operations, onRow, getKey, size, alignCenter, noBorder } =
+    props;
+
+  return (
+    <div className="erda-list">
+      {dataSource.length > 0 ? (
```
Comment (McDaddy) [inline shell/app/common/components/list/index.tsx:27]: no need `> 0`

File: shell/app/common/components/list/index.spec.d.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+declare namespace ERDA_LIST {
+  interface IListItemProps {
+    size?: 'small' | 'middle' | 'large';
+    data: IListData;
+    alignCenter?: boolean;
+    noBorder?: boolean;
+    operations: IOperation[] | ((data: IListData) => IOperation[]);
+    onRow?: Object;
```
Comment (McDaddy) [inline shell/app/common/components/list/index.spec.d.ts:21]: what does `Object` mean?

File: shell/app/common/components/list/index.spec.d.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    key: string | number;
+  }
+
+  interface IOperation {
+    title: React.ReactNode;
+    key?: string | number;
+    onClick?: ([key]: any) => void;
```
Comment (McDaddy) [inline shell/app/common/components/list/index.spec.d.ts:28]: remove any

File: shell/app/common/components/list/index.spec.d.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    operations?: Obj<CP_COMMON.Operation>;
+    prefixImgCircle?: boolean;
+  }
+
+  interface IIconInfo {
+    icon?: string;
+    text: string;
+    type?: 'success' | 'normal' | 'warning' | 'error';
+    tooltip?: string;
+    operations?: Obj<CP_COMMON.Operation>;
```
Comment (McDaddy) [inline shell/app/common/components/list/index.spec.d.ts:70]: Why `CP_COMMON` here

File: shell/app/common/components/list/index.scss [hunk 5]
```diff
@@ -0,0 +1,160 @@
+.erda-list {
+  &-item {
+    margin-top: 0.5rem;
+    border-bottom: 1px solid $color-border;
+    border-radius: 2px;
+    background-color: $white;
+    box-shadow: 0 2px 4px 0 rgba(85, 77, 104, 0.08);
```
Comment (laojun) [inline shell/app/common/components/list/index.scss:7]: defined color in _color.scss，do not add similar color with exist colors

File: shell/app/common/components/list/index.spec.d.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+declare namespace ERDA_LIST {
+  interface IListItemProps {
```
Comment (laojun) [inline shell/app/common/components/list/index.spec.d.ts:15]: common components don't add the spec.d.ts as config-page

File: shell/app/common/components/list/index.spec.d.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    size?: 'small' | 'middle' | 'large';
```
Comment (daskyrk) [inline shell/app/common/components/list/index.spec.d.ts:16]: use ISize

File: shell/app/modules/cmp/pages/cluster-manage/cluster-list.tsx [hunk 8]
```diff
@@ -452,16 +241,47 @@ const ClusterList = ({ dataSource, onEdit }: IProps) => {
...
-          loading={loadingList || loadingDetail}
-          scroll={{ x: 1500 }}
+        <DiceConfigPage
+          scenarioKey="cmp-cluster-list"
+          scenarioType="cmp-cluster-list"
+          customProps={{
+            list: {
+              op: {
+                clickItem: (_: unknown, record: ICluster) => {
+                  goTo(goTo.pages.cmpClustersDetail, { clusterName: record.title });
```
Comment (laojun) [inline shell/app/modules/cmp/pages/cluster-manage/cluster-list.tsx:252]: better not use record.title,  seems record.meta.name is better?

**Pull request merged**